output/secrethub.i

```diff
@@ -4,14 +4,18 @@
 %}
 %include exception.i
 #include "secrethub.h"
+#include <stdio.h>
 
-%exception {
-    clearErr();
-    $action
-    char *errMsg = checkErr();
-    if(strlen(errMsg)) {
-        SWIG_exception(SWIG_RuntimeError, errMsg);
-    }
+%typemap(in, numinputs=0) char **errMessage (char *temp="") {
+  $1 = &temp;
 }
-extern char *Read(char*);
-extern char *Resolve(char*);
+
+%typemap(argout, canthrow=1) char **errMessage {
```

Comment:
I wonder, since we're pattern matching on the name 'errMessage', wouldn't it be more elegant to store a global error message, such as we did before, so that there wouldn't be a pointer initialized for every function that receives the errorMessage as parameter? (which, so far, are 3, but might become more in the future)

Reply:
Oh, my excuses, I presume this is for the concurrent function calls feature?

Reply:
Yes, it is for allowing concurrent function calls indeed. Don't worry about asking stuff like this, it's better to make sure everything is correct.